pkgs/stdenv/generic/check-meta.nix

```diff
@@ -230,6 +231,8 @@ let
       { valid = false; reason = "unfree"; errormsg = "has an unfree license (‘${showLicense attrs.meta.license}’)"; }
     else if hasBlacklistedLicense attrs then
       { valid = false; reason = "blacklisted"; errormsg = "has a blacklisted license (‘${showLicense attrs.meta.license}’)"; }
+    else if hasLicense attrs && !(licenseCheckPredicate attrs) then
+      { valid = false; reason = "blacklisted"; errormsg = "failed licenseCheckPredicate (‘${showLicense attrs.meta.license}’)"; }
```

Comment:
I decided to reuse the `"blacklisted"` reason string here because I wasn't sure if something else handles these strings elsewhere.  Alternatively we could have a new one like `"license-check-failed"` or something.